docs/core-concepts/queries.md

```diff
@@ -0,0 +1,191 @@
+| `onSuccess` | `(value: T) => void` | Callback function called when the query succeeds. Provides the value `T` returned from the function. |
+| `onError` | `(error: unknown) => void` | Callback function called when the query fails. Provides the error `unknown` thrown by the function. |
+| `interval` | `number` | Time in milliseconds between automatic re-fetches. |
+| `retries` | `number \| Partial<RetryOptions>` | Number of retry attempts or retry configuration. |
+| `tags` | `QueryTags` | Tags for cache invalidation and management. |
+
+### Placeholder
+
+By default a query that hasn't been executed (or hasn't finished executing) will use `undefined` for the `data`. However, this can be overridden with anything you wish.
+
+```ts
+const placeholder: Cat[] = []
+const catsQuery = await query(searchCats, ['Bengal'], { placeholder })
+//     ^? data: Cat[]
+```
+
+### Interval
+
+Automatically re-trigger the function on an interval. Value provided is time in milliseconds.
+
+```ts
+const catsQuery = await query(searchCats, ['Persian'], { interval: 30_000 })
+// automatically calls `searchCats` when query is created, and then again every 30 seconds
+```
+
+### Retries
+
+Automatically re-try the function when an error occurs. By default the query will wait `500` milliseconds between retries.
+
+```ts
+const catsQuery = query(searchCats, ['Siamese'], { retries: 3 })
+// will try to call `searchCats` a total of 3 times, waiting 500 MS between each attempt
+```
+
+Alternatively you can pass in `RetryOptions`, which allows you to specify both a `count` and a `delay`.
+
+```ts
+const retries = { count: 1, delay: 1_000 }
+const catsQuery = query(searchCats, ['Siamese'], { retries: 3 })
+// will retry after 1,000 MS once if initial call fails
+```
+
+### Tags
+
+Tags are used to organize queries and mutations that are related. Tags can be used to access cached values, update cache values, and trigger cache invalidation.
+
+## UseQuery method
+
+The `useQuery` function adds some super useful, vue specific functionality you'd expect from a [composition](https://vuejs.org/guide/extras/composition-api-faq.html).
+
+### Parameters argument is reactive
+
+When parameters for your query function change, the query is automatically updated. 
+
+```ts
+import { useQuery } from '@kitbag/query'
+
+const selectedBreed = ref<string | undefined>()
+
+const catsQuery = useQuery(searchCats, () => [selectedBreed.value])
+```
+
+### Delayed execution
+
+The query is suspended while parameters are `null`, this enables you to create your query even when waiting on required arguments.
+
+```ts
+const selectedBreed = ref<string | undefined>()
+
+const catsQuery = useQuery(searchCats, () => {
+  if (!selectedBreed.value) {
+    return null
+  }
+  
+  return [selectedBreed.value]
+})
+```
+
+You can also setup a query NOT to execute immediately. These queries wait until you either call `execute()`, or change the reactive parameters argument.
```

Comment:
Does changing the parameters execute a query that has immediate: false? Feels like it shouldn't.

Reply:
good catch, I looked again and confirmed in query-preview that you MUST call `execute` to re-enable a query when `immediate` is false